core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/FcJsonRpc.java

```diff
@@ -98,6 +102,41 @@ public boolean filecoinWalletHas(@JsonRpcParam("address") final String address)
     return fcSigners.availableIdentifiers().contains(address);
   }
 
+  @JsonRpcMethod("Filecoin.WalletVerify")
+  public boolean filecoinWalletVerify(
+      @JsonRpcParam("address") final String filecoinAddress,
+      @JsonRpcParam("data") final String dataToVerify,
+      @JsonRpcParam("signature") final FilecoinSignature filecoinSignature) {
+    final Optional<ArtifactSigner> signer = fcSigners.getSigner(filecoinAddress);
+
+    if (signer.isPresent()) {
+      final Bytes bytesToVerify = Bytes.fromBase64String(dataToVerify);
+      final ArtifactSignature artifactSignature = createArtifactSignature(filecoinSignature);
+      return signer.get().verify(bytesToVerify, artifactSignature);
+    } else {
+      throw new FilecoinSignerNotFoundException();
+    }
+  }
+
+  private ArtifactSignature createArtifactSignature(final FilecoinSignature filecoinSignature) {
+    final Bytes signature = Bytes.fromBase64String(filecoinSignature.getData());
+    switch (filecoinSignature.getType()) {
+      case SECP_VALUE:
+        final Bytes r = signature.slice(0, 32);
+        final Bytes s = signature.slice(32, 32);
```

Comment:
will this call just return 1 bit?

Reply:
v will be a single byte, needs more than one bit as ethereum adds +27 to recId